### PR TITLE
AsyncHandler: Normalize file requests

### PR DESCRIPTION
**Pull request by Ghabry**
Forgot this one

Test case, created with new gencache from gencache PR:
https://easyrpg.org/play/pr1729/?game=yume_nikki_test

Title screen is renamed to "１２３４５①.xyz" and NASU contains some １s.

Also works because of version check:
https://easyrpg.org/play/pr1729/?game=yume_nikki

Master, fails to load assets in the new normalized one:
https://easyrpg.org/play/master/?game=yume_nikki_test

### CODE REVIEW

**Comment by Ghabry** [comment]
Actually maybe makes sense now to add a version field to gencache.
Because this is a breaking change. See Tools#33

**Comment by Ghabry** [comment]
okay added the version check, now our old index.json files still work ;)

**Comment by fmatthew5876** [comment]
How long do we need to keep around support for the old versions? Are they being used in the wild?

**Comment by Ghabry** [comment]
Well, we use them ;) and there are many pages around who use the web player and when they update it we can't expect from them that they create a new cache.
Most prominent is rmarchiv.tk, has a webplayer for all 2k/2k3 games and this uses a custom php-based index.json generator and somebody would need to submit a patch there first.

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by Ghabry** [comment]
This is blocked by https://github.com/EasyRPG/Tools/pull/40

**Comment by Ghabry** [comment]
sorry I cleaned up my fork and deleted the wrong branch it seems xD

**Comment by fmatthew5876** [comment]
Whats the story with this guy? Still relevant?

**Comment by Ghabry** [comment]
this still requires https://github.com/EasyRPG/Tools/pull/40

**Comment by fdelapena** [comment]
No longer blocked by EasyRPG/Tools#51 :tada:

**Comment by Ghabry** [comment]
I retested now the 3 links above: Still works as expected.

The file "TiTlE/１２３４５①.xyz" is stored in the cache as ``"title":{"123451":"１２３４５①.xyz"`` (+_dirname) and correctly fetched.

File: src/async_handler.cpp [hunk 1]
```diff
@@ -31,19 +33,21 @@
...
+
 
 // When this option is enabled async requests are randomly delayed.
 // This allows testing some aspects of async file fetching locally.
 //#define EP_DEBUG_SIMULATE_ASYNC
 
 namespace {
 	std::unordered_map<std::string, FileRequestAsync> async_requests;
-	std::unordered_map<std::string, std::string> file_mapping;
+	FlatUniqueMultiMap<std::string, std::string> file_mapping;
```
Comment (fmatthew5876) [inline src/async_handler.cpp:48]: Why switch to `FlatUniqueMultiMap`? Do you need multiple key support now?

**Review by fmatthew5876 — APPROVED** [review]

**Comment by Ghabry** [comment]
Retested with the commit removed: Still works (obviously, I just changed the map data structure)

**Review by fdelapena — APPROVED** [review]

**Pull request merged**